site/en/blog/longer-esw-lifetimes/index.md

```diff
@@ -35,11 +36,12 @@ Unfortunately, this behavior did not apply to extension events. Extension events
 
 ## What's changed
 
-As of Chrome 110, all events reset the lidle timer and the idle timeout will not occur if there are pending events. In other words, assuming there are no unexpected interruptions, extension service workers will now stay alive as long as they are actively processing events.
+As of Chrome 110, all events reset the lidle timer and the idle timeout will not occur if there are pending events. In other words, assuming there are no unexpected interruptions, extension service workers will now typically stay alive as long as they are actively processing events. Be aware that:
 
-While this change removes the five minute limit on extension service worker lifetimes, you are still strongly encouraged to design your extensions to be event-based and guard against unexpected termination. Not only should you refrain from keeping your extensions alive indefinitely (which we consider an anti-pattern), you should test your extensions to ensure that they're not doing this unintentionally.
+* The service worker will terminate after 30 seconds if there are no events.
+* The service worker will terminate after 5 minutes if a single request, such as an event or API call, takes longer than 5 minutes to process (with the exception of a native messaging connection).
```

Comment:
I assume this is only the case for pending async calls, correct? Or will synchronous operations be terminated as well?

//cc @rdcronin

Reply:
I don't know. Maybe Devlin does.

Reply:
New wording from Oliver.

Reply:
Oliver didn't like the new wording and he makes some compelling arguments:
* >It won't terminate after 30 seconds if you make certain API calls for example. So it would need to be "The service worker will terminate after 30 seconds if there are no events or API calls to extension APIs and there isn't a native messaging connection, regardless of whether you're running synchronous or asynchronous code." I realise we touch on that elsewhere in the article but I think we should make sure if people quote a specific bullet point it isn't wrong.
* >I feel like synchronous and asynchronous was sort of implied with how we had it before, and calling it out actually adds confusion because then it makes me stop to think about what those are.

Reply:
I'm not sure what the latest wording is but I think something like this would be fine:

> * The service worker will terminate after 30 seconds of inactivity (receiving an event or calling an extension API resets this timer).
> * The service worker will terminate after 5 minutes if a single request, such as an event or API call, takes longer than 5 minutes to process
>
> Some APIs like native messaging provide a strong keep-alive which cancel both of these timers.

And maybe change "be aware that" to "specifically" or something? Since it's not a bad thing.

Reply:
+1 to Oliver's suggestions